src/aap_eda/core/management/commands/scheduler.py

```diff
@@ -149,4 +152,47 @@ def handle(self, *args, **options) -> None:
         add_startup_jobs(scheduler)
         add_periodic_jobs(scheduler)
         add_cron_jobs(scheduler)
-        super().handle(*args, **options)
+        # We are going to start our own loop here to catch exceptions which
+        # might be coming from a redis cluster and retrying things.
+        while True:
+            try:
+                super().handle(*args, **options)
+            except (
+                redis.exceptions.TimeoutError,
+                redis.exceptions.ClusterDownError,
+                redis.exceptions.ConnectionError,
+            ) as e:
+                # If we got one of these exceptions but are not on a Cluster go
+                # ahead and raise it normally.
+                if type(self.connection) is not DABRedisCluster:
+                    raise
+
+                # There are a lot of different exceptions that inherit from
+                # ConnectionError.  So we need to make sure if we got that its
+                # an actual ConnectionError. If not, go ahead and raise it.
+                # Note:  ClusterDownError and TimeoutError are not subclasses
+                #        of ConnectionError.
+                if (
+                    issubclass(redis.exceptions.ConnectionError, e)
+                    and type(e) is not redis.exceptions.ConnectionError
```

Comment:
The issubclass usage is reversed.  As written it's checking that ConnectionError is a subclass of the exception instance.  This allows subclasses of ConnectionError to pass through rather than be raised. This possibly relates to the disconnect logic question.
Further, issubclass documentation indicates that it is a check on classes not between instances and classes.  So it's use here would be incorrect regardless.
If we're intending to raise ConnectionError subclasses an 'isinstance(e, ConnectionError) and type(e) is not ConnectionError' would do so.